Library/TestHelpers/TestCase.swift

```diff
@@ -33,6 +33,9 @@ internal class TestCase: XCTestCase {
     var calendar = Calendar(identifier: .gregorian)
     calendar.timeZone = TimeZone(identifier: "GMT")!
 
+    self.optimizelyClient = MockOptimizelyClient()
+      |> \.features .~ [OptimizelyFeature.consentManagementDialogEnabled.rawValue: true]
+
```

Comment:
Can you remove this from the base `TestCase` and put it back into the `override setUp` for the individual test class? I was thinking about this [method](https://developer.apple.com/documentation/xctest/xctestcase/set_up_and_tear_down_state_in_your_tests#3909114).

Reply:
Create a class level variable `MockOptimizelyClient` in the `override class setup` func in `PledgeViewModelTests` for example, and then update it to `false` for whatever flags you need to be setup, then update the individual cases that need to be true as you've already done.

Reply:
right. I was trying to avoid that because there are a few test files and it was easier to add it here. And then remove it once the feature is fully released. Removing it from one place would also be easier. What do you think about that?

Reply:
Its a good point. I'll go ahead and add a class level variable to all of the test files that are failing

Reply:
Its a good point. We are making an effort to audit and clean up our flags as we think about moving to firebase as a manager though. There are a few different test files that have multiple tests that are failing which feels like a lot of code changes for this. 

I can make a ticket to remove as well so I don't forget, but I can definitely go through and add class level variables too so all of our testcases aren't impacted

Reply:
If it's not too much work, let's use the class level variables. I now we're trying to figure out Optimizely replacements and then yes we'll go ahead and rip out Optimizely everywhere for the replacement but that work isn't quite finalized _yet_.

I think the original creators intent of `TestCase` is not to add customizations of `AppEnvironment`.
We'll see that `withEnvironment(` in `XCTestCase+AppEnvironment` is intended to be customized for every test case.
Likely when we add the replacement for `Optimizely`, we'll follow this same approach and add it to the `withEnvironment(` function and customize it in every test case that needs it.

It shouldn't be too much boilerplate right? Because we're updating the _class_ level `mockOptimizelyClient` not in every test. The only tests that need to change are the the ones that require a `true` flag. I understand its probably four different files, but it still feels like the tradeoff is worth it and sticks to existing conventions. Also when we have this new replacement for Optimizely, then those files are easy to change with the new dependency because we just go through AppEnvironment.

My feel for the app is that it's very consistent in its' patterns and conventions, which ends up being super super nice later when you go to refactor something like `AppEnvironment` out, and then you don't need to worry that `TestCase` specifically needs to also get an update for one dependency....because `AppEnvironment` contains all the dependencies.

Reply:
Ok, looking into this now, it's fine, just do it the way you have in `TestCase`, I'll approve this. The class level update doesn't really seem to work either.